messaging/src/main/java/org/axonframework/commandhandling/distributed/DistributedCommandBus.java

```diff
@@ -177,25 +160,27 @@ public <C, R> void dispatch(@Nonnull CommandMessage<C> command,
         CommandMessage<? extends C> interceptedCommand = intercept(command);
         MessageMonitor.MonitorCallback messageMonitorCallback = messageMonitor.onMessageIngested(interceptedCommand);
         Optional<Member> optionalDestination = commandRouter.findDestination(interceptedCommand);
-        if (optionalDestination.isPresent()) {
-            Member destination = optionalDestination.get();
-            try {
+        Span span = spanFactory.createDispatchSpan(() -> "DistributedCommandBus.dispatch", command).start();
+        try (SpanScope ignored = span.makeCurrent()) {
+            if (optionalDestination.isPresent()) {
+                Member destination = optionalDestination.get();
                 connector.send(destination,
                                interceptedCommand,
                                new MonitorAwareCallback<>(callback, messageMonitorCallback));
-            } catch (Exception e) {
-                messageMonitorCallback.reportFailure(e);
-                destination.suspect();
-                callback.onResult(interceptedCommand, asCommandResultMessage(
-                        new CommandDispatchException(DISPATCH_ERROR_MESSAGE + ": " + e.getMessage(), e)
-                ));
+            } else {
+                throw new NoHandlerForCommandException(
```

Comment:
Through this change, the `NoHandlerForCommandException` is wrapped in a `CommandDispatchException`, while dispatching didn't even start.

Reply:
changed